ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/TransactionBubble.qml

```diff
@@ -2,123 +2,204 @@ import QtQuick 2.3
-    property bool outgoing: true
-    property string state: "addressReceived"
-    property int timestamp: 1598454756329
-
-    id: root
-    width: 170
-    height: childrenRect.height
-    radius: 16
-    color: Style.current.background
-    border.color: Style.current.border
-    border.width: 1
-
-    StyledText {
-        id: title
-        color: Style.current.secondaryText
-        text: outgoing ? qsTr("↑ Outgoing transaction") : qsTr("↓ Incoming transaction")
-        font.weight: Font.Medium
-        anchors.top: parent.top
-        anchors.topMargin: Style.current.halfPadding
-        anchors.horizontalCenter: parent.horizontalCenter
-        font.pixelSize: 13
+import "../../../Wallet/data"
+
+Item {
+    property var commandParametersObject: {
+        try {
+            return JSON.parse(commandParameters)
+        } catch (e) {
+            console.error('Error parsing command parameters')
+            console.error('JSON:', commandParameters)
+            console.error('Error:', e)
+            return {
+                id: "",
+                fromAddress: "",
+                address: "",
+                contract: "",
+                value: "",
+                transactionHash: "",
+                commandState: 1,
+                signature: null
+            }
+        }
     }
-
-    Item {
-        id: valueContainer
-        height: tokenText.height + fiatText.height
-        anchors.top: title.bottom
-        anchors.topMargin: 4
-        anchors.right: parent.right
-        anchors.rightMargin: 12
-        anchors.left: parent.left
-        anchors.leftMargin: 12
-
-        Image {
-            id: tokenImage
-            source: `../../../../img/tokens/${root.token}.png`
-            width: 24
-            height: 24
-            anchors.verticalCenter: parent.verticalCenter
+    property var token: {
+        if (commandParametersObject.contract === "") {
+            return {
+                symbol:   "ETH",
+                name:     "Ethereum",
+                address:  Constants.zeroAddress,
+                decimals: 18,
+                hasIcon: true
+            }
         }
 
-        StyledText {
-            id: tokenText
-            color: Style.current.text
-            text: `${root.tokenAmount} ${root.token}`
-            anchors.left: tokenImage.right
-            anchors.leftMargin: Style.current.halfPadding
-            font.pixelSize: 22
+        let count = walletModel.defaultTokenList.items.count
+        for (let i = 0; i < count; i++) {
+            let token = walletModel.defaultTokenList.items.get(i)
+            if (commandParametersObject.contract === token.address) {
+                return token
+            }
+        }
+        return {}
+    }
+    property string tokenAmount: {
+        if (!commandParametersObject.value) {
+            return "0"
         }
+        try {
+            return Utils.divideByDecimals(commandParametersObject.value, token.decimals)
```

Comment:
Instead of doing this conversion on the QML side (ie string math haha), what do you think about converting the `commandParameters.value` on the nim side with `libstatus/utils/wei2eth`?

Reply:
Are you telling me we had these functions during all that time... 😭